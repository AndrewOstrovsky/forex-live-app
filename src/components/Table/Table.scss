@import '../../styles/variables.scss';
@import '../../styles/mixins.scss';

.c-table {
    border-collapse: collapse;
    margin: 25px 0;
    font-size: 0.9em;
    font-family: sans-serif;
    width: 100%;
    box-shadow: 0 0 20px rgba(0, 0, 0, 0.15);

    &-row {
        background: #fff;
        border-bottom: 1px solid #dddddd;

        &:nth-of-type(even) {
            background-color: #f3f3f3;
        }

        &:last-of-type {
            border-bottom: 2px solid $primaryColor;
        }
    }

    &-row-heading {
        background-color: $primaryColor;
        color: #fff;
        text-align: left;
        font-weight: bold
    }

    &-cell {
        font-size: 12px;
        padding: 12px 15px;
        width: 25%;
        @include md {
            font-size: 14px;
        }
        @include lg {
            font-size: 16px;
        }
    }
}