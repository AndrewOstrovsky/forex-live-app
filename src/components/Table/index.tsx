import React from 'react'
import './Table.scss'

import classNames from 'classNames'

import { IDataItem } from '../../helpers/searchData/types'

interface ITableProps {
    data: IDataItem[]
}

const Table = ({ data }: ITableProps) => {
    const baseClass = 'c-table'
    const cellClassName = `${baseClass}-cell`

    return (
        <table className={baseClass}>
            <thead>
                <tr className={`${baseClass}-row-heading`}>
                    <th className={cellClassName}>Pair</th>
                    <th className={cellClassName}>Bid</th>
                    <th className={cellClassName}>Ask</th>
                    <th className={cellClassName}>Change %</th>
                </tr>
            </thead>
            <tbody>
            {data && data.map((item: IDataItem) => {
                return (
                    <tr key={item.ticker} className={`${baseClass}-row`}>
                        <td className={cellClassName}>{item.ticker}</td>
                        <td className={cellClassName}>{item.bid}</td>
                        <td className={cellClassName}>{item.ask}</td>
                        <td className={classNames(cellClassName,{
                            [`${cellClassName}--positive`]: item.changes > 0,
                            [`${cellClassName}--negative`]: item.changes < 0
                        })}>{item.changes && item.changes.toFixed(2)}%</td>
                    </tr>
                )
            })}
            </tbody>
        </table>
    )
}

export default Table