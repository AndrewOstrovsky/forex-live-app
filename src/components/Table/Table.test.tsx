import React from 'react';

import renderer from 'react-test-renderer';
import Table from '.';

const testData =[
    {
        "ticker" : "EUR/USD",
        "bid" : "1.21218",
        "ask" : "1.21218",
        "changes" : -0.20006421814409003,
    }, {
        "ticker" : "USD/JPY",
        "bid" : "104.188",
        "ask" : "104.188",
        "changes" : 0.33609722743863457,
  }
]

it('renders correctly', () => {
  const component = renderer
    .create(<Table data={testData}/>)
    .toJSON();
  expect(component).toMatchSnapshot();
});