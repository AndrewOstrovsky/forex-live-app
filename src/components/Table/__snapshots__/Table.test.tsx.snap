// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<table
  className="c-table"
>
  <thead>
    <tr
      className="c-table-row-heading"
    >
      <th
        className="c-table-cell"
      >
        Pair
      </th>
      <th
        className="c-table-cell"
      >
        Bid
      </th>
      <th
        className="c-table-cell"
      >
        Ask
      </th>
      <th
        className="c-table-cell"
      >
        Change %
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      className="c-table-row"
    >
      <td
        className="c-table-cell"
      >
        EUR/USD
      </td>
      <td
        className="c-table-cell"
      >
        1.21218
      </td>
      <td
        className="c-table-cell"
      >
        1.21218
      </td>
      <td
        className="c-table-cell c-table-cell--negative"
      >
        -0.20
        %
      </td>
    </tr>
    <tr
      className="c-table-row"
    >
      <td
        className="c-table-cell"
      >
        USD/JPY
      </td>
      <td
        className="c-table-cell"
      >
        104.188
      </td>
      <td
        className="c-table-cell"
      >
        104.188
      </td>
      <td
        className="c-table-cell c-table-cell--positive"
      >
        0.34
        %
      </td>
    </tr>
  </tbody>
</table>
`;
